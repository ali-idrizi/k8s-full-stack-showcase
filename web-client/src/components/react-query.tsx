import { reactQueryConfig } from '@/configs'
import { DehydratedState, Hydrate, QueryClient, QueryClientProvider } from '@tanstack/react-query'
import React, { useState } from 'react'

type ReactQueryProps = {
  dehydratedState?: DehydratedState
}

export const ReactQuery: React.FC<React.PropsWithChildren<ReactQueryProps>> = ({
  dehydratedState,
  children,
}) => {
  const [queryClient] = useState(() => new QueryClient(reactQueryConfig))

  return (
    <QueryClientProvider client={queryClient}>
      <Hydrate state={dehydratedState}>{children}</Hydrate>
    </QueryClientProvider>
  )
}
