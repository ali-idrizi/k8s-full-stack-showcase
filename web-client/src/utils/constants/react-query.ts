export const QUERY_KEY = {
  AUTH: 'auth',
  TODO: 'todo',
  TODO_LISTS: 'todo-lists',
  TODO_LIST: 'todo-list',
} as const

export const MUTATION_KEY = {
  LOGIN: 'login',
  LOGOUT: 'logout',
  REFRESH_TOKEN: 'refresh-token',
  REGISTER: 'register',
  TODO_LIST_CREATE: 'todo-list-create',
  TODO_LIST_UPDATE: 'todo-list-update',
  TODO_LIST_DELETE: 'todo-list-delete',
}
