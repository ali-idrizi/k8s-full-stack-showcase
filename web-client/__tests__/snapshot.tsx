import Home from '@/pages/index'
import { QueryClient, QueryClientProvider } from '@tanstack/react-query'
import { render } from '@testing-library/react'

it('renders homepage unchanged', () => {
  const { container } = render(
    <QueryClientProvider client={new QueryClient()}>
      <Home auth={{ userId: 'test-user-id', needsRefresh: false }} />
    </QueryClientProvider>,
  )
  expect(container).toMatchSnapshot()
})
