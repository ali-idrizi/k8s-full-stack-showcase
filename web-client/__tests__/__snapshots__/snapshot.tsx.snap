// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders homepage unchanged 1`] = `
<div>
  <div
    class="chakra-stack css-4m2dmu"
  >
    <h2
      class="chakra-heading css-a3kn27"
    >
      Create and manage your
       
      <span
        class="chakra-text css-epvm6"
      >
        <mark
          class="css-tlrssp"
        >
          tasks
        </mark>
         and 
        <mark
          class="css-tlrssp"
        >
          todos
        </mark>
      </span>
    </h2>
    <h2
      class="chakra-heading css-1x6wkgp"
    >
      Create lists with your tasks and organize your day better!
    </h2>
    <div
      class="chakra-stack css-1jin9mk"
    >
      <a
        class="chakra-button css-1os6a1l"
        href="/login"
      >
        Login
      </a>
      <a
        class="chakra-button css-1h77yvi"
        href="/register"
      >
        Register
      </a>
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;
